crates/router/src/connector/paypal/transformers.rs

```diff
@@ -0,0 +1,490 @@
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum PaymentSourceItem {
+    Card(CardRequest),
+}
+
+#[derive(Debug, Serialize)]
+pub struct PaypalPaymentsRequest {
+    intent: PaypalPaymentIntent,
+    purchase_units: Vec<PurchaseUnitRequest>,
+    payment_source: Option<PaymentSourceItem>,
+}
+
+fn get_address_info(address: Option<&api_models::payments::Address>) -> Option<Address> {
+    address.and_then(|add| {
+        add.address.as_ref().map(|a| Address {
+            country_code: a.country.clone().unwrap_or_default(),
+            address_line_1: a.line1.clone(),
+            postal_code: a.zip.clone(),
+        })
+    })
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for PaypalPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        match item.request.payment_method_data.clone() {
+            api_models::payments::PaymentMethodData::Card(ref ccard) => {
+                let intent = match item.request.is_auto_capture() {
+                    true => PaypalPaymentIntent::Capture,
+                    false => PaypalPaymentIntent::Authorize,
+                };
+                let amount = OrderAmount {
+                    currency_code: item.request.currency.to_string().to_uppercase(),
+                    value: item.request.amount.to_string(),
+                };
+                let reference_id = item.attempt_id.clone();
+
+                let purchase_units = vec![PurchaseUnitRequest {
+                    reference_id,
+                    amount,
+                }];
+                let card = item.get_card()?;
+                let expiry = Some(card.get_expiry_date_as_yyyymm("-")).map(Secret::new);
+
+                let payment_source = Some(PaymentSourceItem::Card(CardRequest {
+                    billing_address: get_address_info(item.address.billing.as_ref()),
+                    expiry,
+                    name: (ccard.card_holder_name.clone()),
+                    number: Some(ccard.card_number.clone()),
+                    security_code: Some(ccard.card_cvc.clone()),
+                }));
+
+                Ok(Self {
+                    intent,
+                    purchase_units,
+                    payment_source,
+                })
+            }
+            _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, PartialEq)]
+pub struct PaypalAuthUpdateRequest {
+    grant_type: String,
+    client_id: String,
+    client_secret: String,
+}
+impl TryFrom<&types::RefreshTokenRouterData> for PaypalAuthUpdateRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::RefreshTokenRouterData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            grant_type: "client_credentials".to_string(),
```

Comment:
change this to enum if we can pass multiple grant_type

Reply:
No, We only use one grant_type, no multiple required